scripts/build_containers.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import os
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from pathlib import Path
+from typing import List
+
+from jsonargparse import ArgumentParser
+from spython.main import Client
+
+# Define the directory where the projects are located
+BASE_DIR: Path = Path(__file__).resolve().parent.parent / "projects"
+
+# List of all available project names
+PROJECTS: List[str] = [x.name for x in BASE_DIR.iterdir() if x.is_dir()]
+
+
+def build_container(project_name: str, container_root: Path) -> str:
+    project_path: Path = BASE_DIR / project_name
+    container_path: Path = container_root / f"{project_name}.sif"
+
+    if not container_path.exists():
+        out = (
+            f"Apptainer definition file for {project_name} "
+            "does not exist. Skipping build."
+        )
+        return out
+
+    try:
+        Client.build(
+            image=str(container_path),
+            recipe=str(project_path / "apptainer.def"),
+            sudo=False,
+            options=["--force"],
+        )
+        return f"Successfully built container for {project_name}"
+    except Exception as e:
+        return f"Failed to build container for {project_name}: {e}"
+
+
+def build(projects: List[str], container_root: Path, max_workers: int) -> None:
+    if not container_root:
+        logging.info("Container root path is not set.")
+        return
+
+    with ProcessPoolExecutor(max_workers=max_workers) as executor:
+        futures = {
+            executor.submit(build_container, project, container_root): project
+            for project in projects
+            if project in PROJECTS
```

Comment:
Potentially should raise an error if any of `projects` aren't in `PROJECTS`. Otherwise, a typo when specifying which projects to build could cause that container to not be rebuilt without it being obvious